### PR TITLE
Adding TLS support for plugins

### PR DESCRIPTION
**Pull request by marcintao**
- change go version for CI to 1.7; need go 1.7 to support loading system root CA certs

### CODE REVIEW

File: v1/plugin/plugin.go [hunk 1]
```diff
@@ -78,27 +82,140 @@ type StreamCollector interface {
...
+// server, with TLS optionally turned on.
+func makeGRPCCredentials(m *meta) (creds credentials.TransportCredentials, err error) {
+	var config *tls.Config
+	if !m.TLSEnabled {
+		config = &tls.Config{
+			InsecureSkipVerify: true,
+		}
+	} else {
+		cert, err := tls.LoadX509KeyPair(m.CertPath, m.KeyPath)
+		config = tlsSetup.makeTLSConfig()
```
Comment (katarzyna-z) [inline v1/plugin/plugin.go]: It seems that it doesn't make sense to make TLS Config if `tls.LoadX509KeyPair` returns error. Please consider it.

File: v1/plugin/session.go [hunk 2]
```diff
@@ -32,17 +30,28 @@ type Arg struct {
...
+	osArgs := libInputOutput.readOSArgs()
+	var paramStr = osArgs[1]
+	if paramStr == "" {
+		paramStr = "{}"
 	}
-	err := json.Unmarshal([]byte(os.Args[1]), pluginArg)
+	err := json.Unmarshal([]byte(paramStr), pluginArg)
 	if err != nil {
-		return err
+		return nil, err
```
Comment (katarzyna-z) [inline v1/plugin/session.go]: Consider adding of  `pluginArg` to error message, it should be easier to debug user's issues.

File: v1/plugin/session.go [hunk 3]
```diff
@@ -32,17 +30,28 @@ type Arg struct {
...
+	TLSEnabled bool
 }
 
 // getArgs returns plugin args or default ones
-func getArgs() error {
+func getArgs() (*Arg, error) {
 	pluginArg := &Arg{}
-	if os.Args[1] == "" {
-		return nil
+	osArgs := libInputOutput.readOSArgs()
```
Comment (katarzyna-z) [inline v1/plugin/session.go]: Please check number of plugin's arguments

Comment (marcintao) [inline v1/plugin/session.go]: done

File: v1/plugin/session.go (hunk 2 above)
Comment (marcintao) [inline v1/plugin/session.go]: done

File: v1/plugin/plugin.go (hunk 1 above)
Comment (marcintao) [inline v1/plugin/plugin.go]: done

File: v1/plugin/grpc_test.go [hunk 4]
```diff
@@ -22,17 +22,76 @@ limitations under the License.
...
 	"net"
+	"strings"
 	"sync"
 	"testing"
+	"time"
 
 	"golang.org/x/net/context"
 	"google.golang.org/grpc"
 
+	"reflect"
```
Comment (katarzyna-z) [inline v1/plugin/grpc_test.go]: add this to the first group of imports

File: v1/plugin/grpc_tls_test.go [hunk 5]
```diff
@@ -0,0 +1,554 @@
...
+	"math/rand"
+	"os"
+	"strings"
+	"testing"
+	"time"
+
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
+
+	"sync/atomic"
```
Comment (katarzyna-z) [inline v1/plugin/grpc_tls_test.go]: move this import to the first group of imports

File: v1/plugin/plugin_test.go [hunk 6]
```diff
@@ -25,9 +25,14 @@ import (
 	"errors"
 	"fmt"
 	"net"
+	"strings"
 	"testing"
 	"time"
 
+	"crypto/tls"
+
+	"encoding/json"
```
Comment (katarzyna-z) [inline v1/plugin/plugin_test.go]: add these two imports to the first group of imports

File: v1/plugin/session.go [hunk 7]
```diff
@@ -6,10 +6,8 @@ import (
 	"log"
 	"net"
 	"net/http"
-	"os"
-	"time"
-
 	"net/http/pprof"
+	"time"
```
Comment (katarzyna-z) [inline v1/plugin/session.go]: add these two imports to the first group of imports

File: v1/plugin/grpc_test.go (hunk 4 above)
Comment (marcintao) [inline v1/plugin/grpc_test.go]: done

File: v1/plugin/grpc_tls_test.go (hunk 5 above)
Comment (marcintao) [inline v1/plugin/grpc_tls_test.go]: done

File: v1/plugin/plugin_test.go (hunk 6 above)
Comment (marcintao) [inline v1/plugin/plugin_test.go]: done

**Pull request merged**